List Tables: Add support for Screen Options

obenland opened the pull request:
<img width="1582" height="1031" alt="image" src="https://github.com/user-attachments/assets/e7a6ba94-f7d0-46f8-b0da-f380cad0bc33" />


## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Introduces a `Screen_Options` class to manage per-page settings for followers and following list tables. 
* Updates table classes and templates to use global table instances and per-page options
* Hooks screen options into the admin menu load actions for improved user customization.

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to the Follower list page and open screen options.
* Play with the options and make sure the table corresponds correctly.

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Followers and Following list tables now support Columns and Pagination screen options.

</details>

obenland committed:
List Tables: Add support for Screen Options

Introduces a Screen_Options class to manage per-page settings for followers and following admin tables. Updates table classes and templates to use global table instances and per-page options, and hooks screen options into the admin menu load actions for improved user customization.

matticbot committed:
Add changelog

obenland commented:
Follow-ups:
* Make screen options work with Following list template, after #1909.
* Move all existing screen options business into the new class.

obenland commented on includes/table/class-following.php:
```diff
@@ -80,14 +80,10 @@ public function get_sortable_columns() {
-
 		$this->process_action();
-		$this->_column_headers = array( $columns, $hidden, $this->get_sortable_columns() );
 
```
This should be handled in `WP_List_Table::get_column_info()`.

pfefferle commented:
It seems throw a warnings on `/wp-admin/options-general.php?page=activitypub&tab=followers`

<img width="2391" height="440" alt="Screenshot 2025-07-14 at 17 52 22" src="https://github.com/user-attachments/assets/226f3c21-269a-4112-a061-fc146c5dba0c" />

pfefferle committed:
Merge branch 'trunk' into add/hidden-columns

obenland committed:
Merge branch 'trunk' into add/hidden-columns

pfefferle committed:
Set default status

pfefferle committed:
add missing functions

pfefferle committed:
revert globals

pfefferle committed:
Add screen options

obenland committed:
Revert "revert globals"

This reverts commit f96f8da7a220a6e9e36259488782f393e08eb755.

obenland committed:
Revert "add missing functions"

This reverts commit 1a5d9793b0f29e1a81148ceb6789fcb52539a4f4.

obenland committed:
Refactor admin menu and screen options handling

Replaces inline tab-based screen options logic in Menu::admin_menu with dedicated methods in Admin and Screen_Options classes. Renames screen option methods for consistency and adds a new add_settings_list_options method to centralize tab handling. Improves code organization and maintainability.

obenland committed:
Update following template

obenland committed:
Update function name

obenland committed:
Merge branch 'trunk' into add/hidden-columns

obenland commented:
It doesn't look like we can separate column settings for Follower and Following tables in ActivityPub settings for the blog user, they're based on screen id, which is always the same. It's probably fine, both tables have the same columns.

obenland commented:
@pfefferle This should be ready for a review now

pfefferle commented:
There seem to be an interference on the Following and Followers Table for the Blog-User. If I change the Columns on one list it also applies to the other. Pagination seem to work fine though.

obenland commented:
Right, that's what I meant [in this comment](https://github.com/Automattic/wordpress-activitypub/pull/1925#issuecomment-3074035579). We won't really get around it since screen options are based on the screen, not tabs.

pfefferle commented:
Ah, too many comments 🫣

pfefferle commented:
@obenland is it maybe possible to fake something by overloading the https://developer.wordpress.org/reference/functions/get_hidden_columns/
